include/simdutf/implementation.h

```diff
@@ -84,6 +84,18 @@ simdutf_warn_unused size_t convert_utf8_to_utf16(const char * input, size_t leng
  */
 simdutf_warn_unused size_t convert_valid_utf8_to_utf16(const char * input, size_t length, char16_t* utf16_buffer) noexcept;
 
+/**
+ * Convert valid UTF-8 string into UTF-32 string.
+ *
+ * This function assumes that the input string is valid UTF-8.
+ *
+ * @param input         the UTF-8 string to convert
+ * @param length        the length of the string in bytes
+ * @param utf32_buffer  the pointer to buffer that can hold conversion result
+ * @return the number of written char32_t
+ */
+simdutf_warn_unused size_t convert_valid_utf8_to_utf32(const char * input, size_t length, char32_t* utf32_buffer) noexcept;
```

Comment:
is it possible to add also api function, which validate + convert into temp buffer with fixed size. Allocating full buffer not always good idea.

Reply:
You can convert just a substring: provide a pointer with the desired input length.

Reply:
With substring it requires return pointer of the last valid char. To continue from that, at the next iteration. Also  substring is by default invalid utf8

Reply:
It is probably best if you turn your need into a distinct issue.